Schema and docs clean-up pass (#535)
nf-core/eager: docs/usage.md

- Qualimap is run on each sample, after merging of libraries (i.e. your values will reflect the values of all libraries combined - after being damage trimmed etc.).
- Genotyping will typically be performed on each `sample` independently as normally all libraries will have been merged together. However, if you have a mixture of single-stranded and double-stranded libraries, you will normally need to genotype separately. In this case you **must** give each the SS and DS libraries _distinct_ `Sample_IDs` otherwise you will recieve a `file collision` error in steps such as `sexdeterrmine`, and merge these yourself. We will consider changing this behaviour in the future if there is enough interest.  

#### `--udg_type`

> Defines whether Uracil-DNA glycosylase (UDG) treatment was used to repair DNA damage on the sequencing libraries.

Reviewer: Technically untrue. it doesn't repair, just cleaves, right?

Specify `'none'` if no treatment was performed. If you have partial UDG treated data ([Rohland et al 2016](http://dx.doi.org/10.1098/rstb.2013.0624)), specify `'half'`. If you have complete UDG treated data ([Briggs et al. 2010](https://doi.org/10.1093/nar/gkp1163)), specify `'full'`. When also using PMDtools `'half'` will use a different model for DNA damage assessment in PMDTools. Specify the parameter with `'full'` and the PMDtools DNA damage assessment will use CpG context only. Default: `'none'`.

Not required when using TSV input.

#### `--single_stranded`

Indicates libraries were made with a single stranded protocol.

> Currently only affects MALTExtract, where it will switch on damage patterns calculation mode to single-stranded. Default: false.

Reviewer: Also affects pileupcaller genotyping.

Not required when using TSV input.

#### `--colour_chemistry`

Specifies which Illumina colour chemistry was a library was sequenced with. This informs  whether to perform poly-G trimming (if `--complexity_filter_poly_g` is also supplied). Only 2 colour chemistry sequencers (e.g. NextSeq or NovaSeq) can generate uncertain poly-G tails (due to 'G' being indicated via a no-colour detection). Default is '4' to indicate e.g. HiSeq or MiSeq platforms, which do not require poly-G trimming. Options: 2, 4.

Not required when using TSV input.

#### `--bam`

Specifies the input file type to `--input` is in BAM format. This will automatically also apply `--single_end`.

Not required when using TSV input.

### Additional Input Options

#### `--snpcapture_bed`

If your data is from a library that has undergone SNP capture or enrichment (e.g. human 390K,1240K panels), supply a BED file (3/6 column format) containing the positions of the SNPs on the genome that was in the SNP set of the capture. Supplying this bed file will mean on-target metrics  will be calculated automatically for you.

#### `--run_convertinputbam`

> Allows you to convert a input BAM file back to FASTQ for downstream processing. Note this is required if you need to perform AdapterRemoval and/or polyG clipping.

Reviewer: ```suggestion
Allows you to convert an input BAM file back to FASTQ for downstream processing. Note this is required if you need to perform AdapterRemoval and/or polyG clipping.
```

> You must select either a `--fasta` or `--genome`

We provide various options for indexing of different types of reference genomes. nf-core/eager can index reference genomes for you (with options to save these for other analysis), but you can also supply your pre-made indices.

> Supplying pre-made indices saves time in pipeline execution and is especially advised when running multiple times on the same cluster system for example. You can even add a resource [specific profile](#profile) that sets paths to pre-computed reference genomes, saving even time when specifying these.

Reviewer: ```suggestion
Supplying pre-made indices saves time in pipeline execution and is especially advised when running multiple times on the same cluster system for example. You can even add a resource [specific profile](#profile) that sets paths to pre-computed reference genomes, saving time when specifying these.
```

```bash
--fasta_index 'Mammoth_MT_Krause.fasta.fai'

```

#### `--seq_dict`

If you want to use a pre-existing `picard CreateSequenceDictionary` dictionary file, use this to specify the required `.dict` file for the selected reference genome.

For example:

```bash
--seq_dict 'Mammoth_MT_Krause.dict'
```

#### `--large_ref`

> This parameter is required to be set for large reference genomes. If your reference genome is larger than 3.5GB, the `samtools index` calls in the pipeline need to generate `CSI` indices instead of `BAI` indices to accompensate for the size of the reference genome. This parameter is not required for smaller references (including a human `hg19` or `grch37`/`grch38` reference), but `>4GB` genomes have been shown to need `CSI` indices. Default: off

Reviewer: ```suggestion
This parameter is required to be set for large reference genomes. If your reference genome is larger than 3.5GB, the `samtools index` calls in the pipeline need to generate `CSI` indices instead of `BAI` indices to compensate for the size of the reference genome. This parameter is not required for smaller references (including a human `hg19` or `grch37`/`grch38` reference), but `>4GB` genomes have been shown to need `CSI` indices. Default: off
```

Turns off QualiMap and thus does not compute coverage and other mapping metrics.

### Complexity Filtering Options

> More details on can be seen in the [fastp documentation](https://github.com/OpenGene/fastp)

Reviewer: ```suggestion
More details can be seen in the [fastp documentation](https://github.com/OpenGene/fastp)
```

#### `--pmdtools_max_reads`

The maximum number of reads used for damage assessment in PMDtools. Can be used to significantly reduce the amount of time required for damage assessment in PMDTools. Note that a too low value can also obtain incorrect results. Default: `10000`

### BAM Trimming Parameters

> 

Reviewer: Just below here shouldn't the parameter be `--run_trimbam` and not `--trim_bam`

Specify calling method to use. Options: randomHaploid, randomDiploid, majorityCall. Default: `'randomHaploid'`

#### `--pileupcaller_transitions_mode`

> Specify if genotypes of transition SNPs should be called, set to missing, or excluded from the genotypes respectively. Options: AllSites, TransitionsMissing, SkipTransitions. Default: '`AllSites`'

Reviewer: ```suggestion
Specify if genotypes of transition SNPs should be called, set to missing, or excluded from the genotypes respectively. This only applies to double-stranded libraries, as single-stranded libraries are genotyped in a way that eliminates false transitions being called because of DNA damage. Options: AllSites, TransitionsMissing, SkipTransitions. Default: '`AllSites`'
```

Reviewer: Check for more of these quote and backtick flips just in case. I found some, but may have missed a few more.

#### `--database`

Specify the path to the _directory_ containing your taxonomic classifer's database (malt or kraken).

For Kraken2, it can be either the path to the _directory_ or the path to the `.tar.gz` compressed directory of the Kraken2 database.

#### `--metagenomic_min_support_reads`

Specify the minimum number of reads a given taxon is required to have to be retained as a positive 'hit'.  
For malt, this only applies when `--malt_min_support_mode` is set to 'reads'. Default: `1`.

#### `--percent_identity`

Specify the minimum percent identity (or similarity) a squence must have to the reference for it to be retained. Default is `85`

Only used when `--metagenomic_tool malt` is also supplied

#### `--malt_mode`

Use this to run the program in 'BlastN', 'BlastP', 'BlastX' modes to align DNA and DNA, protein and protein, or DNA reads against protein references respectively.

> respectively. Ensure your database matches the mode. Check the [MALT manual](http://ab.inf.uni-tuebingen.de/data/software/malt/download/manual.pdf) for more details. Default: '`BlastN`'  

Reviewer: ```suggestion
respectively. Ensure your database matches the mode. Check the [MALT manual](http://ab.inf.uni-tuebingen.de/data/software/malt/download/manual.pdf) for more details. Default: `'BlastN'`  
```

Only when `--metagenomic_tool malt` is also supplied

#### `--malt_alignment_mode`

> Specify what alignment algorithm to use. Options are 'Local' or 'SemiGlobal'. Local is a BLAST like alignment, but is much slower. Semi-global alignment aligns reads end-to-end. Default: '`SemiGlobal`'

Reviewer: ```suggestion
Specify what alignment algorithm to use. Options are 'Local' or 'SemiGlobal'. Local is a BLAST like alignment, but is much slower. Semi-global alignment aligns reads end-to-end. Default: `'SemiGlobal'`
```

#### `--malt_min_support_percent`

Specify the minimum number of reads (as a percentage of all assigned reads) a given taxon is required to have to be retained as a positive 'hit' in the RMA6 file. This only applies when `--malt_min_support_mode` is set to 'percent'. Default `0.01`.

Only when `--metagenomic_tool malt` is also supplied

#### `--malt_max_queries`

Specify the maximum number of alignments a read can have. All further alignments are discarded. Default: `100`

Only when `--metagenomic_tool malt` is also supplied

#### `--malt_memory_mode`

> How to load the database into memory. Options are 'load', 'page' or 'map'. 'load' directly loads the entire database into memory prior seed look up, this is slow but compatible with all servers/file systems. 'page' and 'map' perform a sort of 'chunked' database loading, allow seed look up prior entire database loading. Note that Page and Map modes do not work properly not with many remote file-systems such as GPFS. Default is '`load`'.

Reviewer: ```suggestion
How to load the database into memory. Options are 'load', 'page' or 'map'. 'load' directly loads the entire database into memory prior seed look up, this is slow but compatible with all servers/file systems. 'page' and 'map' perform a sort of 'chunked' database loading, allow seed look up prior entire database loading. Note that Page and Map modes do not work properly not with many remote file-systems such as GPFS. Default is `'load'`.
```

#### `--maltextract_filter`

> Specify which MaltExtract filter to use. This is used to specify what types of characteristics to scan for. The default will output statistics on all alignments, and then a second set with just reads with one C to T mismatch in the first 5 bases. Further details on other parameters can be seen in the [HOPS documentation](https://github.com/rhuebler/HOPS/#maltextract-parameters). Options: 'def_anc', 'ancient', 'default', 'crawl', 'scan', 'srna', 'assignment'. Default: '`def_anc`'.

Reviewer: ```suggestion
Specify which MaltExtract filter to use. This is used to specify what types of characteristics to scan for. The default will output statistics on all alignments, and then a second set with just reads with one C to T mismatch in the first 5 bases. Further details on other parameters can be seen in the [HOPS documentation](https://github.com/rhuebler/HOPS/#maltextract-parameters). Options: 'def_anc', 'ancient', 'default', 'crawl', 'scan', 'srna', 'assignment'. Default: `'def_anc'`.
```
